src/main/scala/mesosphere/marathon/upgrade/DeploymentActor.scala

```diff
@@ -136,47 +137,51 @@ private class DeploymentActor(
     promise.future
   }
 
+  @SuppressWarnings(Array("all")) /* async/await */
   def scaleRunnable(runnableSpec: RunSpec, scaleTo: Int,
     toKill: Option[Seq[Instance]],
     status: DeploymentStatus): Future[Unit] = {
     logger.debug("Scale runnable {}", runnableSpec)
 
-    val runningInstances = instanceTracker.specInstancesSync(runnableSpec.id).filter(_.state.condition.isActive)
-
     def killToMeetConstraints(notSentencedAndRunning: Seq[Instance], toKillCount: Int) = {
       Constraints.selectInstancesToKill(runnableSpec, notSentencedAndRunning, toKillCount)
     }
 
-    val ScalingProposition(tasksToKill, tasksToStart) = ScalingProposition.propose(
-      runningInstances, toKill, killToMeetConstraints, scaleTo, runnableSpec.killSelection)
-
-    def killTasksIfNeeded: Future[Unit] = {
-      logger.debug("Kill tasks if needed")
-      tasksToKill.fold(Future.successful(())) { tasks =>
-        logger.debug("Kill tasks {}", tasks)
-        killService.killInstances(tasks, KillReason.DeploymentScaling).map(_ => ())
+    async {
+      val instances = await(instanceTracker.specInstances(runnableSpec.id))
+      val runningInstances = instances.filter(_.state.condition.isActive)
+      val ScalingProposition(tasksToKill, tasksToStart) = ScalingProposition.propose(
+        runningInstances, toKill, killToMeetConstraints, scaleTo, runnableSpec.killSelection)
+
+      def killTasksIfNeeded: Future[Unit] = {
+        logger.debug("Kill tasks if needed")
+        tasksToKill.fold(Future.successful(())) { tasks =>
+          logger.debug("Kill tasks {}", tasks)
+          killService.killInstances(tasks, KillReason.DeploymentScaling).map(_ => ())
+        }
       }
-    }
 
-    def startTasksIfNeeded: Future[Unit] = {
-      tasksToStart.fold(Future.successful(())) { tasksToStart =>
-        logger.debug(s"Start next $tasksToStart tasks")
-        val promise = Promise[Unit]()
-        context.actorOf(TaskStartActor.props(deploymentManager, status, scheduler, launchQueue, instanceTracker, eventBus,
-          readinessCheckExecutor, runnableSpec, scaleTo, promise))
-        promise.future
+      def startTasksIfNeeded: Future[Unit] = {
+        tasksToStart.fold(Future.successful(())) { tasksToStart =>
+          logger.debug(s"Start next $tasksToStart tasks")
+          val promise = Promise[Unit]()
+          context.actorOf(TaskStartActor.props(deploymentManager, status, scheduler, launchQueue, instanceTracker, eventBus,
+            readinessCheckExecutor, runnableSpec, scaleTo, promise))
+          promise.future
+        }
       }
-    }
 
-    killTasksIfNeeded.flatMap(_ => startTasksIfNeeded)
+      await(killTasksIfNeeded.flatMap(_ => startTasksIfNeeded))
```

Comment:
I'm don't mind future monads but I'm not a big fan of mixing them with await/async. Let's just stick with async/await.